src/demux/base-audio-demuxer.ts

```diff
@@ -44,20 +44,21 @@ class BaseAudioDemuxer implements Demuxer {
       this.cachedData = null;
     }
 
-    let id3Data = ID3.getID3Data(data, 0) || [];
-    let offset = id3Data.length;
+    let id3Data: Uint8Array | undefined = ID3.getID3Data(data, 0);
+    let offset = id3Data ? id3Data.length : 0;
     let lastDataIndex;
     let pts;
     const track = this._audioTrack;
     const id3Track = this._id3Track;
-    const timestamp = ID3.getTimeStamp(id3Data);
+    const timestamp = id3Data ? ID3.getTimeStamp(id3Data) : undefined;
     const length = data.length;
 
     if (this.initPTS === null) {
-      this.initPTS = Number.isFinite(timestamp) ? timestamp * 90 : timeOffset * 90000;
+      this.initPTS = (timestamp && Number.isFinite(timestamp)) ? timestamp * 90 : timeOffset * 90000;
```

Comment:
I'm not certain the [Number.isFinite](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite) is still relevant in this case.
The timestamp isn't the result of any dangerous calculation (ie. x / 0).

Perhaps simply replace with `this.initPTS = timestamp ? timestamp * 90 : timeOffset * 90000;`

Reply:
What I wanted to communicate is that we don't want a finite `timestamp` of `0` to result in an `initPTS` based on `timeOffset`.

If `isFinite` requires a number lets just `as` it so that an undefined `timestamp` does result in an `initPTS` based on `timeOffset`.
```suggestion
      this.initPTS = Number.isFinite(timestamp as number) ? timestamp * 90 : timeOffset * 90000;
```

Reply:
I've extracted that line to it's own function in order to add a unit test for that logic.

Not strictly necessary of course.
Added the bang to timestamp, otherwise it was still considered possibly undefined by the typescript compiler.
`Number.isFinite(timestamp as number) ? timestamp! * 90 : timeOffset * 90000;`